test/jdk/java/util/zip/ZipFile/InvalidBytesInEntryNameOrComment.java

```diff
@@ -21,163 +21,304 @@
+    // LOC Header offset for the entry name to be modified
+    private static final int LOC_FILE_HEADER_FILENAME_STARTING_OFFSET = 0x1e;
+    // Buffer used to massage the byte array containing the Zip File
+    private ByteBuffer buffer;
+    // Array used to copy VALID_ZIP into prior to each test run
+    private byte[] zipArray;
 
-    // ZIP file with invalid comment field
-    private Path invalidComment;
-
-    @BeforeTest
-    public void setup() throws IOException {
-        // Create a ZIP file with valid name and comment fields
-        byte[] templateZip = templateZIP();
-
-        // Create a ZIP with a CEN name field containing an invalid byte sequence
-        invalidName = invalidName("invalid-name.zip", templateZip);
+    /**
+     * Byte array representing a valid Zip file prior modifying the CEN/LOC
+     * entry name, CEN entry comment or Zip file comment with an invalid
+     * UTF-8 byte sequence.
+     * See the createZipByteArray method which was used to create the original
+     * Zip file
+     * ----------------#1--------------------
+     * [Central Directory Header]
+     * 0x3a: Signature        : 0x02014b50
+     * 0x3e: Created Zip Spec :       0x14 [2.0]
+     * 0x3f: Created OS       :        0x0 [MS-DOS]
+     * 0x40: VerMadeby        :       0x14 [0, 2.0]
+     * 0x41: VerExtract       :       0x14 [2.0]
+     * 0x42: Flag             :      0x808
+     * 0x44: Method           :        0x8 [DEFLATED]
+     * 0x46: Last Mod Time    : 0x58506664 [Fri Feb 16 12:51:08 EST 2024]
+     * 0x4a: CRC              : 0xd202ef8d
+     * 0x4e: Compressed Size  :        0x3
+     * 0x52: Uncompressed Size:        0x1
+     * 0x56: Name Length      :        0x9
+     * 0x58: Extra Length     :        0x0
+     * 0x5a: Comment Length   :        0xc
+     * 0x5c: Disk Start       :        0x0
+     * 0x5e: Attrs            :        0x0
+     * 0x60: AttrsEx          :        0x0
+     * 0x64: Loc Header Offset:        0x0
+     * 0x68: File Name        :  entryName
+     * 0x71: Comment          : [entryComment]
+     * [Local File Header]
+     * 0x0: Signature   :   0x04034b50
+     * 0x4: Version     :         0x14    [2.0]
+     * 0x6: Flag        :        0x808
+     * 0x8: Method      :          0x8    [DEFLATED]
+     * 0xa: LastMTime   :   0x58506664    [Fri Feb 16 12:51:08 EST 2024]
+     * 0xe: CRC         :          0x0
+     * 0x12: CSize       :          0x0
+     * 0x16: Size        :          0x0
+     * 0x1a: Name Length :          0x9    [entryName]
+     * 0x1c: ExtraLength :          0x0
+     * 0x1e: File Name   : [entryName]
+     * [End Central Directory Header]
+     * 0x7d: Signature   :   0x06054b50
+     * 0x85: Disk Entries:          0x1
+     * 0x87: Total Entries:         0x1
+     * 0x89: CEN Size    :         0x43
+     * 0x8d: Offset CEN  :         0x3a
+     * 0x91: Comment Len :          0xe
+     * 0x93: Comment     :   [ZipFileComment]
+     */
+    public static byte[] VALID_ZIP = {
```

Comment:
I see this byte array encoding of ZIP files is a pattern used across tests. My preference would be to encode this in Base64 or hex, since that would make the consumption by external command line tools easier and the encoded representation somewhat prettier. But no big deal, this might just come down to personal preference. (This isn't human readable anyhow for the normal definition of "human" :)